### PR TITLE
local-up-cluster.sh: support preserving etcd optionally

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

**What this PR does / why we need it**:

Allow `hack/local-up-cluster.sh` to support preserve `ETCD_DIR` when `PRESERVE` is set to non-false value.

usage:
```
$ PRESERVE=true ETCD_DIR=/var/lib/etcd-kube-HEAD .... \
   ./hack/local-up-cluster.sh
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
(None)

**Special notes for your reviewer**:
(None)

**Release note**:

```release-note
None
```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -71,6 +71,8 @@ CLOUD_PROVIDER=${CLOUD_PROVIDER:-""}
 CLOUD_CONFIG=${CLOUD_CONFIG:-""}
 FEATURE_GATES=${FEATURE_GATES:-"AllAlpha=false"}
 STORAGE_BACKEND=${STORAGE_BACKEND:-"etcd3"}
+# preserve persistent stuff such as etcd. you also need to set ETCD_DIR.
+PRESERVE="${PRESERVE:-false}"
```
Comment (ixdy) [inline hack/local-up-cluster.sh]: nit: maybe `PRESERVE_ETCD`? otherwise this is pretty ambiguous.

Comment (AkihiroSuda) [reply hack/local-up-cluster.sh]: updated, thanks for quick review

**Comment by AkihiroSuda** [comment]
@ixdy @spiffxp @lavalamp Any chance to get this merged? :pray:

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -396,8 +398,9 @@ cleanup()
 
   # Check if the etcd is still running
   [[ -n "${ETCD_PID-}" ]] && kube::etcd::stop
-  [[ -n "${ETCD_DIR-}" ]] && kube::etcd::clean_etcd_dir
-
+  if [ "${PRESERVE_ETCD}" == "false" ]; then
```
Comment (ixdy) [inline hack/local-up-cluster.sh]: nit: we prefer `[[ ]]` over `[ ]` for bash tests.

**Comment by ixdy** [comment]
/assign

sorry, this fell off my radar

**Pull request merged**